app/src/main/java/comcmput301f16t01/github/carrier/Users/NewUserActivity.java

```diff
@@ -31,34 +31,36 @@ public void onBackPressed() {
         this.finish();
     }
 
+
     /**
-     * @param v The button (view) the user presses that calls this method.
+     * This is the function that will take us to the next screen.
+     * We will bundle all info entered, check validity and then send it to next activity
+     * @param v the view needed for the function
      */
-    public void submitNewUser(View v) {
+
+    public void nextVehicle(View v) {
         EditText usernameEditText = (EditText) findViewById(R.id.editText_newUsername);
         EditText phoneEditText = (EditText) findViewById(R.id.editText_newPhoneNum);
         EditText emailEditText = (EditText) findViewById(R.id.editText_newEmail);
 
         String username = usernameEditText.getText().toString();
+        String phone = phoneEditText.getText().toString();
+        String email = emailEditText.getText().toString();
 
         // Send new user to UserController
         UserController uc = new UserController();
-        String result = uc.createNewUser(username,
-                emailEditText.getText().toString(),
-                phoneEditText.getText().toString());
+        String result = uc.checkValidInputs(username, email, phone);
 
-        // Ensure that a new user was created
+        // Ensure that result is okay
         if (result == null) {
-            // Save username to file
-            LoginMemory lm = new LoginMemory( this );
-            lm.saveUsername( username );
-
-            // Log in to CARier
-            String welcome = "Welcome to CARrier, " + username + "!";
-            Intent intent = new Intent(NewUserActivity.this, MainActivity.class);
-            Toast.makeText(this, welcome, Toast.LENGTH_SHORT).show();
+            // We will go to a new activity to add vehicle stuff
+            Intent intent = new Intent(NewUserActivity.this, NewVehicleActivity.class);
+            Bundle bundle = new Bundle();
+            bundle.putString("username", username);
+            bundle.putString("phone", phone);
+            bundle.putString("email", email);
+            intent.putExtras(bundle);
             startActivity(intent);
```

Comment:
Make sure you finish this activity before moving on to the next activity. (or at least make sure it is finished before you submit the user and move on to the MainActivity).